api.planx.uk/saveAndReturn/validateSession.ts

```diff
@@ -1,22 +1,32 @@
 import { gql } from "graphql-request";
 import { NextFunction, Request, Response } from "express";
-import * as jsondiffpatch from "jsondiffpatch";
 import {
   adminGraphQLClient as adminClient,
   publicGraphQLClient as publicClient,
 } from "../hasura";
-import { getMostRecentPublishedFlow, getPublishedFlowByDate } from "../helpers";
+import { getSaveAndReturnPublicHeaders } from "./utils";
+import { getMostRecentPublishedFlow } from "../helpers";
 import {
-  getSaveAndReturnPublicHeaders,
-  stringifyWithRootKeysSortedAlphabetically,
-} from "./utils";
-import { normalizeFlow, sortBreadcrumbs } from "@opensystemslab/planx-core";
+  sortBreadcrumbs as normalizedBreadcrumbSort,
+  ComponentType,
+} from "@opensystemslab/planx-core";
 import type {
-  OrderedBreadcrumbs,
   NormalizedCrumb,
-} from "@opensystemslab/planx-core";
-import type { Breadcrumb, LowCalSession, Node } from "../types";
+  OrderedBreadcrumbs,
+  FlowGraph,
+} from "@opensystemslab/planx-core/types/types";
+import type { PublishedFlow, Node, LowCalSession } from "../types";
 
+export interface ValidationResponse {
+  message: string;
+  reconciledSessionData: LowCalSession["data"];
+  alteredNodes?: Node[];
+  removedBreadcrumbIds?: string[];
+}
+
+// TODO - ensure reconciliation handles:
+//  * collected flags
+//  * auto-answered component dependencies like FindProp/Draw/PlanningConstraints
```

Comment:
thanks for cleaning up these stray code comments

tiny nit: this last line refers to separate things: auto answers AND component dependencies, so might be best split onto a second line for clarity